packages/devtools_app/test/standalone_ui/ide_shared/property_editor_test.dart

```diff
@@ -2,80 +2,145 @@
 // found in the LICENSE file.
 
+import 'dart:async';
+
 import 'package:devtools_app/devtools_app.dart';
-import 'package:devtools_app/src/standalone_ui/ide_shared/property_editor/property_editor_sidebar.dart';
+import 'package:devtools_app/src/service/editor/api_classes.dart';
+import 'package:devtools_app/src/standalone_ui/ide_shared/property_editor/property_editor_controller.dart';
+import 'package:devtools_app/src/standalone_ui/ide_shared/property_editor/property_editor_view.dart';
 import 'package:devtools_app_shared/ui.dart';
 import 'package:devtools_app_shared/utils.dart';
 import 'package:devtools_test/devtools_test.dart';
 import 'package:devtools_test/helpers.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
 
 void main() {
-  const propertyEditor = PropertyEditorSidebar();
+  final eventController = StreamController<ActiveLocationChangedEvent>();
+  final eventStream = eventController.stream;
 
-  group('Property Editor input types', () {
-    setUpAll(() {
-      setGlobal(ServiceConnectionManager, FakeServiceConnectionManager());
-      setGlobal(IdeTheme, IdeTheme());
-    });
+  final locationToArgsResult =
+      <(TextDocument, CursorPosition), EditableArgumentsResult>{
+        (textDocument1, activeCursorPosition1): result1,
+        (textDocument2, activeCursorPosition2): result2,
+      };
 
-    testWidgets('string input', (tester) async {
-      await tester.pumpWidget(wrap(propertyEditor));
+  late MockEditorClient mockEditorClient;
+  late PropertyEditorController controller;
+  late PropertyEditorView propertyEditor;
 
-      final stringInput = _findTextFormField('title');
-
-      expect(stringInput, findsOneWidget);
-    });
+  setUpAll(() {
+    setGlobal(ServiceConnectionManager, FakeServiceConnectionManager());
+    setGlobal(IdeTheme, IdeTheme());
+  });
 
-    testWidgets('double input', (tester) async {
-      await tester.pumpWidget(wrap(propertyEditor));
+  setUpAll(() {
+    mockEditorClient = MockEditorClient();
+    when(
+      mockEditorClient.activeLocationChangedStream,
+    ).thenAnswer((_) => eventStream);
 
-      final doubleInput = _findTextFormField('width');
+    controller = PropertyEditorController(mockEditorClient);
+    propertyEditor = PropertyEditorView(controller: controller);
+  });
 
-      expect(doubleInput, findsOneWidget);
-    });
+  group('on cursor location change', () {
+    void waitForEditableArgs(
+      List<EditableArgument> args, {
+      required Future then,
+    }) {
+      controller.editableArgs.addListener(() async {
+        final current =
+            controller.editableArgs.value.map((arg) => arg.name).toList();
+        final expected = args.map((arg) => arg.name).toList();
 
-    testWidgets('bool input', (tester) async {
-      await tester.pumpWidget(wrap(propertyEditor));
+        if (expected.every((arg) => current.contains(arg))) {
```

Comment:
nit: if you want you can use the `collectionEquals` utility with `ordered: false` instead: `if (collectionEquals(current, expected, ordered: false))`

Reply:
oh nice, done!